data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/extension/DefaultGeoIpConfigSupplier.java

```diff
@@ -5,15 +5,29 @@
 
 package org.opensearch.dataprepper.plugins.processor.extension;
 
+import org.opensearch.dataprepper.plugins.processor.extension.databasedownload.GeoIPDatabaseManager;
+
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
 public class DefaultGeoIpConfigSupplier implements GeoIpConfigSupplier {
     private final GeoIpServiceConfig geoIpServiceConfig;
+    private final ReentrantReadWriteLock.ReadLock readLock;
+    private final GeoIPDatabaseManager geoIPDatabaseManager;
 
-    public DefaultGeoIpConfigSupplier(final GeoIpServiceConfig geoIpServiceConfig) {
+    public DefaultGeoIpConfigSupplier(final GeoIpServiceConfig geoIpServiceConfig,
+                                      final GeoIPDatabaseManager geoIPDatabaseManager,
+                                      final ReentrantReadWriteLock.ReadLock readLock
+                                      ) {
         this.geoIpServiceConfig = geoIpServiceConfig;
+        this.geoIPDatabaseManager = geoIPDatabaseManager;
+        this.readLock = readLock;
     }
 
     @Override
     public GeoIPProcessorService getGeoIPProcessorService() {
-        return new GeoIPProcessorService(geoIpServiceConfig);
+        if (geoIpServiceConfig != null)
```

Comment:
Consider making this an `Optional`. Then the code that calls it can look somewhat like:

```
this.service = supplier.getGeoIpProcessorService().orElseThrow(() -> ... existing exception);
```